code/modules/mob/living/simple_animal/bot/hygienebot.dm

```diff
@@ -0,0 +1,356 @@
+			walk_to(src,0)
+			look_for_lowhygiene()	// see if any disgusting fucks are in range
+			if(!mode && auto_patrol)	// still idle, and set to patrol
+				mode = BOT_START_PATROL	// switch to patrol mode
+
+		if(BOT_HUNT)		// hunting for stinkman
+			// if can't reach stinkman for long enough, don't give up, try harder.
+			if(emagged) //lol fuck em up
+				currentspeed = 8
+				start_washing()
+				mad = TRUE
+			else
+				switch(frustration)
+					if(0 to 4)
+						currentspeed = 5
+						stop_washing()
+						mad = FALSE
+					if(4 to INFINITY)
+						currentspeed = 2.5
+						start_washing()
+						mad = TRUE
+			if(target)
+				if(target.loc == loc && isturf(target.loc)) //LADIES AND GENTLEMAN WE GOTEM PREPARE TO DUMP
+					start_washing()
+					if(mad)
+						speak("Well about fucking time you degenerate", "Fucking finally", "Thank god, you finally stopped")
+						playsound(loc, 'sound/effects/hygienebot_angry.ogg', 60, 1)
+						mad = FALSE
+					mode = BOT_SHOWERSTANCE
+				else
+					var/turf/olddist = get_dist(src, target)
+					walk_to(src, target,0, currentspeed)
+					if(mad && prob(60))
+						playsound(loc, 'sound/effects/hygienebot_angry.ogg', 60, 1)
+						speak(pick("Get back here you foul smelling fucker.", "If you don't get back here right now I'm going to give you a fucking vasectomy.", "STOP RUNNING OR I WILL CUT YOUR ARTERIES!", "Just fucking let me clean you you arsehole!", "STOP. RUNNING.", "Either you stop running or I will fucking drag you out of an airlock.", "I just want to fucking clean you you troglodyte.", "If you don't come back here I'll put a green cloud around you cunt."))
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+		if(BOT_SHOWERSTANCE)
+			if(check_purity(target))
+				speak("Enjoy your clean and tidy day!")
+				playsound(loc, 'sound/effects/hygienebot_happy.ogg', 60, 1)
+				back_to_idle()
+				return
+			if(!target)
+				last_found = world.time
+			if(target.loc != loc && !isturf(target.loc))
+				back_to_hunt()
+
+		if(BOT_START_PATROL)
+			look_for_lowhygiene()
+			start_patrol()
+
+		if(BOT_PATROL)
+			look_for_lowhygiene()
+			bot_patrol()
+
+/mob/living/simple_animal/bot/hygienebot/proc/back_to_idle()
+	mode = BOT_IDLE
+	walk_to(src,0)
+	target = null
+	frustration = 0
+	last_found = world.time
+	stop_washing()
+	INVOKE_ASYNC(src, .proc/handle_automated_action)
+
+/mob/living/simple_animal/bot/hygienebot/proc/back_to_hunt()
+	frustration = 0
+	mode = BOT_HUNT
+	stop_washing()
+	INVOKE_ASYNC(src, .proc/handle_automated_action)
+
+/mob/living/simple_animal/bot/hygienebot/proc/look_for_lowhygiene()
+	for (var/mob/living/carbon/human/H in view(7,src)) //Find the NEET
+		if((H.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+		if(!check_purity(H)) //Theyre impure
+			target = H
+			oldtarget_name = H.name
+			speak("Unhygienic client found, Please stand still so I can clean you")
```

Comment:
Unhygienic client found. Please stand still so I can clean you.

Reply:
pls use code suggestions if u can it makes small fixes easier

Reply:
```suggestion
			speak("Unhygienic client found. Please stand still so I can clean you.")
```